enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/AuthFilter.java

```diff
@@ -134,17 +147,36 @@ public boolean handleRequest(RequestContext requestContext) {
             return true;
         }
 
+        boolean authenticated = false;
         boolean canAuthenticated = false;
         for (Authenticator authenticator : authenticators) {
             if (authenticator.canAuthenticate(requestContext)) {
-                canAuthenticated = true;
+                if (!authenticator.getName().contains(APIConstants.API_SECURITY_MUTUAL_SSL_NAME)) {
+                    canAuthenticated = true;
+                }
                 AuthenticationResponse authenticateResponse = authenticate(authenticator, requestContext);
-                if (authenticateResponse.isAuthenticated() && !authenticateResponse.isContinueToNextAuthenticator()) {
+                if (authenticateResponse.isMandatoryAuthentication()) {
+                    authenticated = authenticateResponse.isAuthenticated();
                     setInterceptorAuthContextMetadata(authenticator, requestContext);
-                    return true;
+                }
+                if (!authenticateResponse.isContinueToNextAuthenticator()) {
+                    break;
+                }
+            } else {
+                if (isMutualSSLMandatory && authenticator.getName()
+                        .contains(APIConstants.API_SECURITY_MUTUAL_SSL_NAME)) {
+                    log.debug("mTLS authentication was failed for the request: " + requestContext.getRequestPath());
+                    break;
+                }
+                if (isOAuthBasicAuthMandatory && !authenticator.getName()
+                        .contains(APIConstants.API_SECURITY_MUTUAL_SSL_NAME)) {
+                    authenticated = false;
                 }
             }
         }
```

Comment:
By looking at the above code segment, I feel like the following scenario may break.
application security - optional
mtls- mandatory

send a request with no auth token but valid mtls. -> isn't it gonna fail according to the above logic, where as expected behaviour is to have no auth failure.

Reply:
After iterating through the mtls authenticator, 'authenticated' is set to true. Then with no auth token, the sequence will reach [this](https://github.com/TharukaN17/product-microgateway/blob/57ed21b2724123bbad50354c2830b74531518ff5/enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/AuthFilter.java#L165). Here, non of the if statements will be reached since not an mtls authenticator and application security optional. 'authenticated' remains true. No auth failure.